File: greedy/boats_to_save_people.py
def numRescueBoats(people: list[int], limit: int) -> int:
    ans = 0
    lightest_idx = 0
    heaviest_idx = len(people) - 1
    people.sort()

    while lightest_idx <= heaviest_idx:
        if people[lightest_idx] + people[heaviest_idx] <= limit:
            lightest_idx += 1
        
        heaviest_idx -= 1
        ans += 1
    
    return ans

File: greedy/boats_to_save_people.java
public int numRescueBoats(int[] people, int limit)
{
    int ans = 0;
    int lightest_idx = 0;
    int heaviest_idx = people.length - 1;
    Arrays.sort(people);

    while (lightest_idx <= heaviest_idx)
    {
        if (people[lightest_idx] + people[heaviest_idx] <= limit)
        {
            lightest_idx++;
        }

        heaviest_idx--;
        ans++;
    }

    return ans;
}